### PR TITLE
fixing datetime to handle locale and UTC times

### PR DESCRIPTION
**Pull request by EricGentry045**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,4 +74,4 @@ Commands support placeholders similar to tasks.json.
 * [Source Code](https://github.com/emeraldwalk/vscode-runonsave)
 
 ## License
-[Apache](https://github.com/emeraldwalk/vscode-runonsave/blob/master/LICENSE)
\ No newline at end of file
+[Apache](https://github.com/emeraldwalk/vscode-runonsave/blob/master/LICENSE)
```
Comment (EricGentry045) [inline README.md:77]: This looks like a whitespace change.  I didn't modify this line intentionally.

**Comment by bmingles** [comment]
@EricGentry045 This looks useful. Would you mind providing some example commands in the PR description that I can use to verify the functionality?

**Review by bmingles — CHANGES_REQUESTED** [review]
@EricGentry045 This looks useful. Would you mind providing some example commands in the PR description that I can use to verify the functionality?

Also, I just merged some updates. Will need a rebase or merge.

**Comment by EricGentry045** [comment]
fixed merge conflict

**Comment by EricGentry045** [comment]
Thanks @bmingles.  Here would be a good use case...

`
{
    "emeraldwalk.runonsave": {
        "commands": [
            {
                "match": "\\.java$",
                "cmd": "git add ${file} && git commit -m 'Auto save ${file} at ${datetime}'"
            },
            {
                "match": "\\.cpp$",
                "cmd": "git add ${file} && git commit -m 'Auto save ${file} at ${datetimeUTC}'"
            },
        ]
    }
}
`

**Review by bmingles — CHANGES_REQUESTED** [review]
There's a few issues I'm seeing here.

1. There are compilation errors due to a typo (should be `toLocaleString()`) and the 2nd arg of String.replace needs to be a string.
    image

2. `new Date()` doesn't actually produce a `UTC` format when it's stringified. It's the local time.
3. The naming convention for date tokens should probably be camelCase `dateTime`
4. I don't like how `javascript's` different `toXXX` methods format the locale date and the stringified date differently, so I'd prefer to find a way to format them more consistently, differing only in timezone offset.

I dug into this a bit to figure out how to make things more consistent with the local / UTC formats. Looks like you can use `toLocaleString` but pass a UTC `timeZone` for the UTC version. I'd also recommend introducing a `dateISO` token since that is a common format.

```javascript
const { locale } = new Intl.DateTimeFormat().resolvedOptions();
const now = new Date();

cmdStr = cmdStr.replace(/\${dateTimeISO}/g, now.toISOString());

// locale formats
cmdStr = cmdStr.replace(/\${dateTime}/g, now.toLocaleString(locale));
cmdStr = cmdStr.replace(
  /\${dateTimeUTC}/g,
  now.toLocaleString(locale, { timeZone: 'UTC' }),
);
```

**Pull request closed without merging**